src/CacheFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
```

Comment:
Why a factory? Do you expect several caching repositories?

Reply:
I had originally written it as an instantiable class, and then realized it made sense to only have one application instance so as a quick way to remedy that I added a factory class. I also didn't care to parameterize this to the WebPush instance nor pass it through to the class that builds the headers.